test/tests/neutronics/adaptivity/tests

```diff
@@ -0,0 +1,36 @@
+[Tests]
+  [adaptive_cell]
+    type = CSVDiff
+    input = cell.i
+    csvdiff = cell_out.csv
+    requirement = "The system shall allow problems which contain adaptivity on the mesh mirror for cell tallies."
```

Comment:
What does this test show exactly? Can we see the outlines of a CSG cell become more refined as the mesh changes?

Reply:
This test ensures that problems can run mesh adaptivity when using cell tallies, so it's measuring the increasing number of elements. It's really just a surrogate for a `RunApp` executioner with some extra information attached.